sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-files/ece-file-detail.ts

```diff
@@ -0,0 +1,65 @@
+import { DisbursementValue } from "@sims/sims-db";
+import { END_OF_LINE, round, StringBuilder } from "@sims/utilities";
+import {
+  DATE_FORMAT,
+  ECERequestFileLine,
+  NUMBER_FILLER,
+  RecordTypeCodes,
+  SPACE_FILLER,
+} from "../models/ece-integration.model";
+
+/**
+ * Record of a ECE request file.
+ * The documentation about it is available on the document
+ * 'SIMSSFAS - Institution File layouts In Analysis Folder'.
+ */
+export class ECERequestFileDetail implements ECERequestFileLine {
+  transactionCode: RecordTypeCodes;
+  institutionCode: string;
+  disbursementValues: DisbursementValue[];
+  sin: string;
+  studentLastName: string;
+  studentGivenName: string;
+  birthDate: string;
+  sfasApplicationNumber: string;
+  institutionStudentNumber: string;
+  courseLoad: string;
+  studyStartDate: string;
+  studyEndDate: string;
+  disbursementDate: string;
+
+  getFixedFormat(): string {
+    const records = this.disbursementValues.map((disbursementValue) => {
+      const record = new StringBuilder();
+      record.append(this.transactionCode, 1);
+      record.append(this.institutionCode, 4);
+      record.appendWithStartFiller(
+        disbursementValue.id.toString(),
+        10,
+        NUMBER_FILLER,
+      );
+      record.append(disbursementValue.valueCode, 4);
+      record.appendWithStartFiller(
+        round(disbursementValue.valueAmount),
+        9,
+        NUMBER_FILLER,
+      );
+      record.append(this.sin, 9);
+      record.appendWithEndFiller(this.studentLastName, 25, SPACE_FILLER);
+      record.appendWithEndFiller(this.studentGivenName ?? "", 15, SPACE_FILLER);
+      record.appendDate(this.birthDate, DATE_FORMAT);
+      record.append(this.sfasApplicationNumber, 10);
+      record.appendWithEndFiller(
+        this.institutionStudentNumber,
```

Comment:
what is student number?

Reply:
It is an optional number that the student can provide during application submission that represents his record on that institution.
From our perspective, we just save alongside the student application dynamic data. I believe that this is the first place where it is really used.

Reply:
oh.. okay. so `his record on that institution` does it mean something like his/her roll number? something like that

Reply:
Got it @andrewsignori-aot thanks.
![image](https://user-images.githubusercontent.com/54600590/210881620-5e361848-524c-40ad-8554-482e592f2af9.png)

Reply:
It is an optional filed on the integration file @ann-aot.